php/WP_CLI/Dispatcher/Subcommand.php

```diff
@@ -448,6 +448,33 @@ public function invoke( $args, $assoc_args, $extra_args ) {
 		}
 		WP_CLI::do_hook( "before_invoke:{$cmd}" );
 
+		// Check if `--prompt` arg passed or not.
+		if ( ! empty( $prompted_once ) && true === $prompted_once ) {
+
+			// Unset empty args.
+			$actual_args = $assoc_args;
+			foreach ( $actual_args as $key ) {
+				if ( empty( $actual_args[ $key ] ) ) {
+					unset( $actual_args[ $key ] );
+				}
+			}
+
+			// Update key ( Append `--` to key ).
+			$prompt_args = [];
+			foreach ( $actual_args as $key => $value ) {
+				$prompt_args[ '--' . $key ] = $value;
+			}
+
+			WP_CLI::log(
+				sprintf(
+					'wp %s %s',
+					$cmd,
+					urldecode( http_build_query( $prompt_args, '', ' ' ) )
```

Comment:
Instead of prefixing everything with `--` manually and then using a combination of `http_build_query()` and `urldecode()`, you can just use the helper function `WP_CLI\Utils\assoc_args_to_str()`.